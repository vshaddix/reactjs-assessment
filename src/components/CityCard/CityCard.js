import React, { Component } from 'react';
import LoadingContent from '../LoadingContent/LoadingContent';
import Measurements from '../Measurements/Measurements';
import MeasurementsService from '../../services/MeasurementsService';

class CityCard extends Component {
  constructor(props) {
    super();

    this.measurementsService = new MeasurementsService();
    this.state = {
      city: props.city,
      country: props.country,
      measurements: null
    };
  }

  componentDidMount() {
    this.measurementsService.getMeasurementsForACity(this.state.city, this.state.country).then(result => {
      this.setState({
        measurements: result,
      });
    });
  }

  render() {
    if (!this.state.city) {
      return <LoadingContent linesCount={3} />;
    }

    return (
      <div className="ui card">
        <div className="content">
          <div className="header">{this.state.city}</div>
        </div>
        <div className="content">
          <h4 className="ui sub header">Measurements</h4>
          <Measurements measurements={this.state.measurements} />
        </div>
      </div>
    );
  }
}

export default CityCard;
