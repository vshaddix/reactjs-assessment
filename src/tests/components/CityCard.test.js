import React from 'react';
import { create } from "react-test-renderer";
import CityCard from '../../components/CityCard/CityCard';

jest.mock('../../services/MeasurementsService', () => {
  return function () {
    return {
      getMeasurementsForACity: () => {
        return new Promise(resolve => resolve([]))
      }
    }
  };
});

it('renders (snapshot test)', () => {
  const component = create(<CityCard />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctly when city and country are passed', () => {
  const component = create(<CityCard city={"Sofiq"} country={"Bulgaria"} />);

  expect(component.toJSON()).toMatchSnapshot();
});
