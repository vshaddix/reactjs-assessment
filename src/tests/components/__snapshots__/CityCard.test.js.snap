// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (snapshot test) 1`] = `
<div
  className="ui placeholder"
>
  <div
    className="line"
  />
  <div
    className="line"
  />
  <div
    className="line"
  />
</div>
`;

exports[`renders correctly when city and country are passed 1`] = `
<div
  className="ui card"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Sofiq
    </div>
  </div>
  <div
    className="content"
  >
    <h4
      className="ui sub header"
    >
      Measurements
    </h4>
    <div
      className="ui placeholder"
    >
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
      <div
        className="line"
      />
    </div>
  </div>
</div>
`;
